node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/NodeAlerterTest.java

```diff
@@ -0,0 +1,151 @@
+package com.yahoo.vespa.hosted.provision.maintenance;
+
+import com.yahoo.config.provision.NodeResources;
+import com.yahoo.config.provision.NodeType;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.*;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+import static org.junit.Assert.*;
+
+/**
+ * @author mgimle
+ */
+public class NodeAlerterTest {
+    private NodeAlerterTester tester;
+    private NodeAlerter alerter;
+
+    @Before
+    public void setup() {
+        tester = new NodeAlerterTester();
+        alerter = tester.makeNodeAlerter();
+    }
+
+    @Test
+    public void testWithRealData() throws IOException {
+        String path = "./src/test/resources/zookeeper_dump.json";
+
+        tester.cleanRepository();
+        tester.restoreNodeRepositoryFromJsonFile(Paths.get(path));
+        NodeAlerter.HostFailurePath failurePath = alerter.worstCaseHostLossLeadingToFailure();
+        if (failurePath != null) {
+            assertTrue(tester.nodeRepository.getNodes(NodeType.host).containsAll(failurePath.hostsCausingFailure));
+        } else fail();
+    }
+
+    @Test
+    public void testWithGeneratedData() {
```

Comment:
Split the different test cases you have in this method into multiple named tests